src/miner.cpp

```diff
@@ -199,11 +199,35 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn, CWallet* pwallet,
             double dPriority = 0;
             CAmount nTotalIn = 0;
             bool fMissingInputs = false;
+            uint256 txid = tx.GetHash();
             for (const CTxIn& txin : tx.vin) {
                 //zerocoinspend has special vin
                 if (tx.IsZerocoinSpend()) {
                     nTotalIn = tx.GetZerocoinSpent();
-                    break;
+
+                    //Give a high priority to zerocoinspends to get into the next block
+                    //Priority = (age^6+100000)*amount - gives higher priority to zpivs that have been in mempool long
+                    //and higher priority to zpivs that are large in value
+                    double nTimeSeen = GetAdjustedTime();
+                    double nConfs = 100000;
+                    double dPriorityPrev = dPriority;
+
+                    auto it = mapZerocoinspends.find(txid);
+                    if (it != mapZerocoinspends.end()) {
+                        nTimeSeen = static_cast<double>(it->second);
+                    } else {
+                        //for some reason not in map, add it
+                        mapZerocoinspends[txid] = GetAdjustedTime();
+                        nTimeSeen = GetAdjustedTime();
```

Comment:
Nit: Not sure `nTimeSeen` needs to be set here since it was initialized to this value a few lines above. And the map value could be set to `nTimeSeen` in order to avoid a call to `GetAdjustedTime()`. No idea how it's doing performance wise.

Reply:
I refactored the code a bit, and forgot to delete this line, good catch.

Fixed and force pushed.